integration/connector_test.go

```diff
@@ -0,0 +1,430 @@
+
+	dittoClient.Subscribe(func(requestID string, msg *protocol.Envelope) {
+		if msg.Path != fmt.Sprintf("/features/%s/inbox/messages/%s", featureName, commandName) {
+			suite.T().Logf("unexpected command: %s", msg.Path)
+			return
+		}
+
+		headers := protocol.NewHeaders(protocol.WithCorrelationID(msg.Headers.CorrelationID()),
+			protocol.WithContentType("application/json"))
+
+		value, ok := msg.Value.(string)
+		if !ok {
+			suite.T().Errorf("unexpected message payload: %v, %T", msg.Value, msg.Value)
+			return
+		}
+
+		response := fmt.Sprintf(commandResponseFormat, value)
+
+		reply := &protocol.Envelope{
+			Topic:   msg.Topic,
+			Headers: headers,
+			Path:    strings.Replace(msg.Path, "/inbox/", "/outbox/", 1),
+			Value:   response,
+			Status:  http.StatusNoContent,
+		}
+
+		if err := dittoClient.Reply(requestID, reply); err != nil {
+			suite.T().Error("failed to send response:", err)
+		}
+	})
+
+	suite.dittoClient = dittoClient
+	suite.mqttClient = mqttClient
+	suite.cfg = cfg
+	suite.thingCfg = thingCfg
+
+	suite.thingURL = fmt.Sprintf("http://%s:%d/api/2/things/%s", cfg.DittoHost, cfg.DittoPort, thingCfg.DeviceID)
+	suite.featureURL = fmt.Sprintf("%s/features/%s", suite.thingURL, featureName)
+}
+
+func (suite *ConnectorSuite) TearDownSuite() {
+	if _, err := suite.doRequest(http.MethodDelete, suite.featureURL); err != nil {
+		suite.T().Logf("error while deleting test feature: %v", err)
+	}
+
+	suite.dittoClient.Disconnect()
+	suite.mqttClient.Disconnect(uint(suite.cfg.MqttQuiesceMs))
+}
+
+func TestConnectorSuite(t *testing.T) {
+	suite.Run(t, new(ConnectorSuite))
+}
+
+func (suite *ConnectorSuite) TestConnectionStatus() {
+	statusURL := fmt.Sprintf("%s/features/ConnectionStatus/properties/status", suite.thingURL)
+	body, err := suite.doRequest("GET", statusURL)
+	require.NoError(suite.T(), err, "connection status property should be available")
+
+	type connectionStatus struct {
+		ReadySince time.Time `json:"readySince"`
+		ReadyUntil time.Time `json:"readyUntil"`
+	}
+
+	status := &connectionStatus{}
+	err = json.Unmarshal(body, status)
+	require.NoError(suite.T(), err, "connection status should be parsed")
+
+	suite.T().Logf("%+v", status)
+	suite.T().Logf("current time: %v", time.Now())
+
+	delta := int64(suite.cfg.TimeDeltaMs)
+	assert.Less(suite.T(), status.ReadySince.UnixMilli(), time.Now().UnixMilli()+delta, "readySince should be BEFORE current time")
+	assert.Greater(suite.T(), status.ReadySince.UnixMilli(), time.Now().UnixMilli()-delta, "readyUntil should be AFTER current time")
+}
+
+func (suite *ConnectorSuite) TestCommand() {
+	ws, err := suite.newWSConnection()
+	defer ws.Close()
+
+	cmd := commandPayload{}
+	namespace := model.NewNamespacedIDFrom(suite.thingCfg.DeviceID)
+	cmd.Topic = fmt.Sprintf("%s/%s/things/live/messages/%s", namespace.Namespace, namespace.Name, commandName)
+	cmd.Path = fmt.Sprintf("/features/%s/inbox/messages/%s", featureName, commandName)
+	cmd.Value = "request"
```

Comment:
The request can be Ditto compliant.

Reply:
According to the [API spec](https://www.eclipse.org/ditto/basic-messages.html#payload), Ditto does not care what the message payload is.

Reply:
@e-grigorov What do you suggest we should do here? We can leave as-is for now.

Reply:
@dimiter-todorov-georgiev - as this approach doesn't break Ditto spec, no updates are required. I'll resolve the conversation.